src/compliment/sources/ns_mappings.clj

```diff
@@ -101,17 +101,36 @@
                                  arglists :function
                                  :else :var)
                      :ns (str (or (:ns var-meta) ns))}
-              (and arglists(:arglists *extra-metadata*))
+              (and arglists (:arglists *extra-metadata*))
               (assoc :arglists (apply list (map pr-str arglists)))
 
               (and doc (:doc *extra-metadata*))
               (assoc :doc (generate-docstring var-meta)))))))))
 
+(defn- resolve-var
+  "Examples:
+  \"map\" => #'clojure.core/map
+
+  ;; Resolves aliases
+  \"some-alias/some-fn\" => #'some.full.ns/some-fn
+
+  ;; Ignores preceding literals (i.e. @, #', ')
+  \"#'map\" => #'clojure.core/map
+  \"@#'some-alias/db\" => #'some.full.ns/db
+  "
+  [symbol-str ns]
+  (let [strip-literals                    #(->> % (re-find #"^[@#']*(.+)") last)
+        [alias-or-ns-str symbol-name-str] ((juxt namespace name) (symbol (strip-literals symbol-str)))
+        symbol-ns                         (when alias-or-ns-str
+                                            (str (or (find-ns (symbol alias-or-ns-str))
+                                                     (get (ns-aliases ns) (symbol alias-or-ns-str)))))]
+    (ns-resolve ns (symbol symbol-ns symbol-name-str))))
```

Comment:
I wonder how much of this has to be done manually since `ns-resolve` already handles a lot of cases. E.g.:

```clj
user=> (ns-resolve *ns* (symbol "set/difference"))
#'clojure.set/difference
user=> (ns-resolve *ns* (symbol "clojure.set/difference"))
#'clojure.set/difference
```

Reply:
Woha!, stripping lots of code 👍🏻